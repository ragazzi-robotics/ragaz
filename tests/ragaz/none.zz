class Bar:
    value: uint
    def __init__(self, v: uint):
        self.value = v


def return_nullable(i: int) -> Bar:
    return Bar(i) if i > 2 else None


def passing_nullable(i: int):
    if i is None:
        print("no number")
    else:
        print(i)


def main():

    var num: int = None
    passing_nullable(num)
    num = 5
    passing_nullable(num)

    var obj: Bar = None
    print(obj is None)
    obj = Bar(2)
    print(obj is None)

    obj = return_nullable(1)
    if obj is None:
        print("no object")
    else:
        print(obj.value)

    obj = return_nullable(3)
    if obj is None:
        print("no object")
    else:
        print(obj.value)
