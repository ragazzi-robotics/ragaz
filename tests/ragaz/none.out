no number
5
True
False
no object
3
